src/arith/rewrite_simplify.cc

```diff
@@ -473,6 +487,9 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const SubNode* op) {
         floordiv(x - y, c1) * c3 - x * c2, (0 - floormod(x - y, c1) - y) * c2,
         c1.Eval()->value != 0 && c3.Eval()->value == c1.Eval()->value * c2.Eval()->value);
 
+    TVM_TRY_RECURSIVE_REWRITE(floordiv(x + 1, 2) - floormod(x, 2), floordiv(x, 2));
+    TVM_TRY_RECURSIVE_REWRITE(floordiv(x, 2) - floormod(x - 1, 2), floordiv(x - 1, 2));
```

Comment:
Like above, `floormod(x - 1, 2)` -> `1 - x % 2`, then it seems apply `x // 2 + x % 2` rule could achieve `(x + 1) // 2 - 1`

Reply:
Updated here as well.  The related rule on line 490 with `floordiv(x + 1,2)` is kept, because there isn't an equivalent change to the argument of `floordiv`.